indico/modules/events/editing/controllers/backend/timeline.py

```diff
@@ -128,8 +135,19 @@ def _process(self):
         args = parser.parse({
             'files': EditingFilesField(self.event, self.contrib, self.editable_type, required=True)
         })
-
-        create_new_editable(self.contrib, self.editable_type, session.user, args['files'])
+        service_url = editing_settings.get(self.event, 'service_url')
+        initial_state = InitialRevisionState.new if service_url else InitialRevisionState.ready_for_review
+
+        editable = create_new_editable(self.contrib, self.editable_type, session.user, args['files'], initial_state)
+        db.session.commit()
+        if service_url:
+            try:
+                service_handle_editable(editable)
+            except ServiceRequestFailed as exc:
+                delete_editable(editable)
+                db.session.commit()
+                logger.exception('Failed to contact service. Got: %r', exc.message)
+                raise BadRequest(_('Failed while submitting for editing. The associated service is not available.'))
```

Comment:
`BadRequest` does not fit here since it's not the client's fault. `ServiceUnavailable` (503) would be more appropriate.

Reply:
Sounds good. I made it `BadRequest` initially because I was considering errors coming from corrupted client input.